tests/cache/models/samples/updateSampleAttributes.js

```diff
@@ -0,0 +1,346 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/cache/models/samples/updateSampleAttributes.js
+ */
+'use strict'; // eslint-disable-line strict
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const model = require('../../../../cache/models/samples');
+const updateSampleAttributes = model.updateSampleAttributes;
+
+describe('tests/cache/models/samples/updateSampleAttributes.js >', () => {
+  let clock;
+
+  before(() => {
+    clock = sinon.useFakeTimers(new Date('2018-10-30T20:24:37.053Z').getTime());
+  });
+
+  after(() => clock.restore());
+
+  it('no sampObj', () => {
+    const qb = {
+      name: '___Subject|___ThreeHours',
+      value: 2,
+      subjectId: '323ef102-3b64-458d-b134-29d2cd840bb2',
+      aspectId: '1979b734-bece-446e-9cb1-03917f60788e',
+    };
+    const samp = null;
+    const asp = {
+      id: '1979b734-bece-446e-9cb1-03917f60788e',
+      isDeleted: '0',
+      relatedLinks: [],
+      tags: [],
+      isPublished: 'true',
+      name: '___ThreeHours',
+      timeout: '3H',
+      valueType: 'NUMERIC',
+      criticalRange: [0, 0],
+      warningRange: [1, 1],
+      infoRange: [2, 2],
+      okRange: [3, 3],
+      updatedAt: '2018-10-29T20:24:37.044Z',
+      createdAt: '2018-10-29T20:24:37.044Z',
+    };
+    updateSampleAttributes(qb, samp, asp);
+    expect(qb).to.deep.equal({
+      name: '___Subject|___ThreeHours',
+      value: 2,
+      subjectId: '323ef102-3b64-458d-b134-29d2cd840bb2',
+      aspectId: '1979b734-bece-446e-9cb1-03917f60788e',
+      previousStatus: 'Invalid',
+      statusChangedAt: '2018-10-30T20:24:37.053Z',
+      status: 'Invalid',
```

Comment:
This is invalid because the value is not a string, right? That's not very clear, is there a reason for writing the test this way?

Reply:
In this test, it doesn't actually matter what the value is -- this test is really about exercising the block of code we enter when this is a brand new sample, i.e. we don't have a "prev" sample to compare to.